### PR TITLE
FEATURE: Allow category moderators to set auto deletion topic timers

### PR DESCRIPTION
**Pull request by xfalcox**

### CODE REVIEW

File: spec/requests/topics_controller_spec.rb [hunk 1]
```diff
@@ -4013,6 +4013,32 @@ def extract_post_stream
         expect(response.status).to eq(403)
         expect(response.parsed_body["error_type"]).to eq('invalid_access')
       end
+
+      it "raises an error if the user sets a delete timer" do
```
Comment (pmusaraj) [inline spec/requests/topics_controller_spec.rb:4017]:
```suggestion
      it "does not allow TL4 to set a delete timer" do
```

File: spec/requests/topics_controller_spec.rb [hunk 2]
```diff
@@ -4013,6 +4013,32 @@ def extract_post_stream
...
+        post "/t/#{topic.id}/timer.json", params: {
+          time: 10,
+          status_type: 'delete'
+        }
+
+        expect(response.status).to eq(403)
+        expect(response.parsed_body["error_type"]).to eq('invalid_access')
+      end
+
+      it "raises an error if the user sets a delete_replies timer" do
```
Comment (pmusaraj) [inline spec/requests/topics_controller_spec.rb:4030]:
```suggestion
      it "raises an error if a TL4 user sets a delete_replies timer" do
```

**Review by pmusaraj — APPROVED** [review]
Looks good! Can we also add a spec that ensures category moderators can indeed add auto deletion topic timers?

File: app/assets/javascripts/discourse/tests/acceptance/topic-edit-timer-test.js [hunk 3]
```diff
@@ -305,6 +311,48 @@ acceptance("Topic - Edit timer", function (needs) {
...
+    await click(".admin-topic-timer-update button");
+
+    const timerType = selectKit(".select-kit.timer-type");
+
+    await timerType.expand();
+
+    assert.ok(!timerType.rowByValue("delete_replies").exists());
+  });
+
+  test("Category Moderator can auto-delete", async function (assert) {
```
[the remark this answers is not in the document]
Comment (xfalcox) [reply app/assets/javascripts/discourse/tests/acceptance/topic-edit-timer-test.js:342]: It's super weird indeed!!! Took me hours to figure this out, but in our fixtures the `can_delete` key is default `true` for almost every single topic in there. Not sure why, but that is why the test cases are a bit weird and it took me many hours to find why stuff for TL4 was broken in tests but working in prod.

[the remark this answers is not in the document]
Comment (pmusaraj) [reply app/assets/javascripts/discourse/tests/acceptance/topic-edit-timer-test.js:342]: Interesting! We should follow-up and flip `can_delete` to `false` and see how many specs fail.

**Comment by xfalcox** [comment]
> Looks good! Can we also add a spec that ensures category moderators can indeed add auto deletion topic timers?

done in a2335a5

**Review by pmusaraj — APPROVED** [review]

**Pull request merged**